lab-shelly/test/server-test.js

```diff
@@ -0,0 +1,257 @@
+'use strict';
+
+const server = require('../server');
+const chai = require('chai');
+const http = require('chai-http');
+const expect = chai.expect;
+
+chai.use(http);
+
+describe('Server module', function () {
+  let app;
+  before(done => {
+    app = server.listen(8000);
+    done();
+  });
+
+  after(done => {
+    app.close();
+    done();
+  });
+
+  describe('POST method', function() {
+    describe('a properly formatted request', function() {
+      it('should return a 200 response' , (done) => {
+        chai.request(server)
+        .post('/api/cat')
+        .send({name: 'milo', mood: 'happy'})
+        .end((err, res) => {
+          expect(res).to.have.status(200);
+          done();
+        });
+      });
+      it('should return a response object', (done) => {
+        chai.request(server)
+        .post('/api/cat')
+        .send({name: 'milo', mood: 'happy'})
+        .end((err, res) => {
+          expect(res).to.be.an('object');
+          done();
+        });
+      });
+    });
+    describe('a request with no body provided or invalid body', function() {
+      it('should return a 400 error response', (done) => {
+        chai.request(server)
+        .post('/api/cat')
+        .send()
+        .end((err, res) => {
+          expect(res).to.have.status(400);
+          done();
+        });
+      });
+    });
+    describe('an unregistered route', function() {
+      it('should return a 404 error ', (done) => {
+        chai.request(server)
+        .get('/api/dog')
+        .end((err, res) => {
+          expect(res).to.have.status(404);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('GET method', function() {
```

Comment:
So I tried running your tests on my local machine, and it seems some of these tests failed.